---
title: 'Linkage disequilibrium - understanding the evolutionary past and mapping the medical future... A review'
date: 2022-03-26
permalink: /posts/2022/03/blog-post-4/
tags:
  - Penn Medicine Biobank
  - Nature reviews genetics
  - Linkage disequilibrium
  - Genetics
---

In this blog post, I'll cover a paper published by Montgomery Slatkin in Nature Reviews genetics in 2008, entitled "Linkage disequilibrium - understanding the evolutionary past and mapping the medical future". It can be found on [Nature's website here](https://www.nature.com/articles/nrg2361) and free on [Pubmed here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5124487/). 

During this review, I also consulted several youtube videos and blogs to provide more information about linkage disequilibrium (LD). 

Background
------
Linkage disequilibrium is a confusing topic, yet it's fundamental to understanding how genes are passed down and affect phenotype. This is a paper written to Montgomery Slatkin in Nature Reviews Genetics, in a series called "Fundamental Concepts in Genetics". He describes the basic concepts of linkage disequilibrium and elaborates on how they can be used to study population genetics. Montgomery Slatkin is an American Biologist and professor at University of Berkeley.

Linkage disequilibrium explained
------
Linkage disequilibrium, at its core, is the nonrandom association of alleles at different loci. Let's unpack this

Defining terms:
- We refer to a locus as a physical location of a genes or genetic marker on the chromosome, notated by a letter e.g. "A" or "B"
- We refer to an allele as a particular version of multiple versions of a locus or gene
- For a locus with 2 alleles, we notate alleles using upper or lower case, e.g. "A" or "a"
- For a locus with more than 2 alleles, we notate using numbers, e.g. "<img src="https://render.githubusercontent.com/render/math?math=A_1">", "<img src="https://render.githubusercontent.com/render/math?math=A_2">", "<img src="https://render.githubusercontent.com/render/math?math=A_3">", etc.
- We notate the frequency of an allele "A" in a population with "<img src="https://render.githubusercontent.com/render/math?math=P_A">"

Remember that for a human with 2 of each chromosome, it's possible to have 0, 1, or 2 copies of allele "a"

For perfect Mendelian genetics, we can remind ourselves of the equation to calculate frequencies for bi-allelic genes as:

![Math](https://latex.codecogs.com/svg.latex?\Large&space;1=P_A+P_a) 

![Math](https://latex.codecogs.com/svg.latex?\Large&space;1=P_AP_A+2P_AP_a+P_aP_a)

If we continue to use Mendelian gentics and we look at just one chromosome, we can calculate the frequency of particular combinations of alleles (A or a) at 2 different loci (A and B)

<img src="https://render.githubusercontent.com/render/math?math=P_A_B =P_AP_B">

<img src="https://render.githubusercontent.com/render/math?math=P_A_b=P_AP_b">

<img src="https://render.githubusercontent.com/render/math?math=P_a_B=P_aP_B">

<img src="https://render.githubusercontent.com/render/math?math=P_a_b=P_aP_b">

Discussion and limitations
------

Take home points
-------
