---
title: 'Recreating all Variant IDs for each SNP rsid in our dataset"
date: 2022-04-30
permalink: /posts/2022/04/blog-post-13/
tags:
  - Penn Medicine Biobank
  - Cancer risk
  - GWAS
  - Polygenic risk score
  - PRS
---

Very briefly, I had challenges this weekend using the plink --extract function. This is because --extract requires Variant ID in the form of chr_pos_ref_alt. In GWAS tables, they provide chr_pos, with many also providing the alt. However, ref allele is not given.

Below, I explain how I recreated a list of all variant IDs for my SNPs.

First, I needed to use the plink --get-snplist for all of my chromosome .vcfs. The code for this was the following:

> module load plink
> 
> bsub -e chr1.e -o chr1.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr1_GL.vcf.gz --write-snplist -out chr1
bsub -e chr2.e -o chr2.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr2_GL.vcf.gz --write-snplist -out chr2
bsub -e chr3.e -o chr3.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr3_GL.vcf.gz --write-snplist -out chr3
bsub -e chr4.e -o chr4.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr4_GL.vcf.gz --write-snplist -out chr4
bsub -e chr5.e -o chr5.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr5_GL.vcf.gz --write-snplist -out chr5
bsub -e chr6.e -o chr6.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr6_GL.vcf.gz --write-snplist -out chr6
bsub -e chr7.e -o chr7.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr7_GL.vcf.gz --write-snplist -out chr7
bsub -e chr8.e -o chr8.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr8_GL.vcf.gz --write-snplist -out chr8
bsub -e chr9.e -o chr9.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr9_GL.vcf.gz --write-snplist -out chr9
bsub -e chr10.e -o chr10.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr10_GL.vcf.gz --write-snplist -out chr10
bsub -e chr11.e -o chr11.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr11_GL.vcf.gz --write-snplist -out chr11
bsub -e chr12.e -o chr12.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr12_GL.vcf.gz --write-snplist -out chr12
bsub -e chr13.e -o chr13.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr13_GL.vcf.gz --write-snplist -out chr13
bsub -e chr14.e -o chr14.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr14_GL.vcf.gz --write-snplist -out chr14
bsub -e chr15.e -o chr15.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr15_GL.vcf.gz --write-snplist -out chr15
bsub -e chr16.e -o chr16.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr16_GL.vcf.gz --write-snplist -out chr16
bsub -e chr17.e -o chr17.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr17_GL.vcf.gz --write-snplist -out chr17
bsub -e chr18.e -o chr18.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr18_GL.vcf.gz --write-snplist -out chr18
bsub -e chr19.e -o chr19.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr19_GL.vcf.gz --write-snplist -out chr19
bsub -e chr20.e -o chr20.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr20_GL.vcf.gz --write-snplist -out chr20
bsub -e chr21.e -o chr21.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr21_GL.vcf.gz --write-snplist -out chr21
bsub -e chr22.e -o chr22.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chr22_GL.vcf.gz --write-snplist -out chr22
bsub -e chrX.e -o chrX.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chrX_GL.vcf.gz --write-snplist -out chrX
bsub -e chrY.e -o chrY.o plink --vcf /project/PMBB/PMBB-Release-2020-2.0/Exome/pVCF/GL_by_chrom/PMBB-Release-2020-2.0_genetic_exome_chrY_GL.vcf.gz --write-snplist -out chrY

Next, I downloaded all the .snplist files onto my laptop and loaded them into R. I then did the following

> chr1snps <- read.csv(file=file.choose(), header = FALSE, sep = "\t")
> 
> chr1snps$chrpos=sub("([^_]+_[^_]+).*", "\\1", chr1snps$V1)

The above command allowed creating a table with now 2 columns. One with header "V1" containing the full variant ID. The second with header "chrpos" which contains only chr_pos.

> chrposlist <- read.csv(file=file.choose(), header = FALSE, sep = "\t")
> colnames(chrposlist)<-("chrpos")
> 

