---
title: 'Genetic imputation and TOPMed r2'
date: 2022-04-18
permalink: /posts/2022/04/blog-post-8/
tags:
  - Penn Medicine Biobank
  - Genetics
  - Imputation
  - TOPMed
---

I have never understood genetic imputation. This blog post serves to describe the concept of genetic imputation and how it's carried out using TOPMed.

Genetic Imputation:
------
Genetic imputation refers to the inference of unobserved genetic data. Put simply, it is when we estimate missing data.

We don't evaluate the entire genome. Instead, we evaluate the genome through single nucleotide polymorphism (SNP) sites. For example, Penn Medicine Biobank implements Illumina "Global Screening Array" microchips which sample approximately ~627,168 SNPs across the entire genome. 

Although this sounds like a lot, it's not. The human genome consists of approximately 3.2 Billion nucleotides. This means that an Illumina GSA V2 chip will evaluate 627,168/3,200,000,000=0.02% of the human genome. This converts to measuring 1 of every 5,000 nucleotides.

Why don't we just sequence the entire genome? 
- Sequencing the whole genome of each individual in the study is often too costly, so only a subset of the genome can be measured
- Not everyone has the technology to sequence the entire genome - many platforms only evaluate specific parts, such as the Illumina sequencing above

So, we haven't analyzed the entire genome. Instead, we analyzed 1 of every 5,000 nucleotides. Now what?

Next, we can use imputation to estimate the sequence that goes in between our 1 of every 5,000 nucleotides. Below is a visual representation of what this woulud entail. As you can see, the 

![Image](https://oliver-clark.github.io/images/1567076168351.png)





**Sources for this blog post include:**
- [Wikipedia Imputation](https://en.wikipedia.org/wiki/Imputation_(genetics))
- 
