---
title: 'Reviewing PRS literature Part 2'
date: 2022-12-12
permalink: /posts/2022/12/blog-post-21/
tags:
  - Penn Medicine Biobank
  - Cancer risk
  - GWAS
  - Polygenic risk score
  - PRS
---

Today, I'm reviewing several more papers on PRS literature, this time in slightly more depth.

1. [On cross-ancestry cancer polygenic risk scores](https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1009670)

As noted in the last post, it may be possible to use cross-ancestry PRS if other criteria are filled. This paper showcases PRS built based on predominantly European populations applied to non-European populations (African, East Asian, and South Asian). What is striking in the figure below.  the consistent right shift of the PRS distributions in cases compared to controls with each ancestry group. This is called a Violin plot

![image1](https://oliver-clark.github.io/images/journal.pgen.1009670.g001.PNG)

Furthermore, the table below shows that in each ancestry, the differentiation of the top 10% is still significant.

![image2](https://oliver-clark.github.io/images/journal.pgen.1009670.t002.PNG)

Of note, the creation of these PRS was in two forms: 

GPRS (GWAS-hits based PRS) which is very similar as the current PRS used in our lab. It sounds like they don't use the 1000 Genomes project as a reference when calculating LD. Risk = beta(i) * SNP(i) + ... + beta(l) * Birthyear + beta(m) * Array # + beta(n) * PC1-10

CS-PRS (Continuous Shrinkage PRS) which was created using a software package in R. They use the 1000 Genomes project as a reference for LD (just like us), but also apply a few different filters (MAF>1%) which I'm not sure if we do. Still, Risk = beta(i) * SNP(i) + ... + beta(l) * Birthyear + beta(m) * Array # + beta(n) * PC1-10.
