---
title: 'At atlas of genetic correlations across human diseases and traits... A review'
date: 2022-03-25
permalink: /posts/2022/03/blog-post-3/
tags:
  - Penn Medicine Biobank
  - Nature genetics
  - GWAS
  - SNPs
---

In this posting, I'll review an analytical paper in Nature genetics entitled "An atlas of genetic correlations across human disease and traits", by Bulik-Sullivan et al. published 2015. It can be found [here on Nature's website](https://www.nature.com/articles/ng.3406) and [here on PubMed](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4797329/).

Background
------
Previously, it has been difficult to estimate genetic correlations between complex traits and diseases using GWAS (Genome wide association studies). Why is this the case? This is because GWAS studies use statistics to identify a SNP that is **really** significantly associated with a trait or disease. But in traits such as height, or diseases such as asthma, there are **many** individual genes that each contribute a small aspect towards the phenotype. For these complex traits and diseases, our conventional statistics don't work. They look for SNPs that obtain genome-wide significance (strongly correlated), but not all associated gene loci are found because many do not reach statistical significance.

This paper provides a new technique which enables estimating correlations between complex traits and diseases. This technique is referred to as cross-trait LD (Linkage disequilibrium) score regression.

The main benefit of this technique is that it may be applied when GWAS studies are available, but individual-level genotypic data is not. These genetic correlations are challenging to evaluate when GWAS studies group together genetic data. To contrast, when individual-level genetic data and phenotypes are available, the statistical tests that could be used are known as REstricted Maximum Likelihood (REML) and polygenic scores. These statistical techniques are discussed elsewhere.

PICOTS
------


Results
------
The purpose of Figure 1 below is to demonstrate that the LDSC (LD scores) don't differ from conventional REML scores. Within each category, LDSC bars are not significantly different to the REML bars.

![Figure 1](https://oliver-clark.github.io/images/tileshop(1).jpg)

The purpose of Figure 2 below is to show that there are some complex traits/diseases that were significantly associated according to the cross-trait LD score regression. These links are marked with a black asterix.
![Figure 2](https://oliver-clark.github.io/images/tileshop.jpg)

I don't yet understand Table 2.
![Table 2](https://oliver-clark.github.io/images/Screenshot_11.jpg)

Discussion and limitations
------

Take home points
------
