---
title: 'Large-scale association analysis identifies new lung cancer susceptibility... A review'
date: 2022-03-23
permalink: /posts/2022/03/blog-post-2/
tags:
  - Penn Medicine Biobank
  - Nature genetics
  - Lung cancer
  - GWAS
---

In the upcoming days, I'm going to review background literature on the topic of my future research. 

Today, I'm going to evaluate a paper published in June 2017 in Nature Genetics Entitled "Large-scale association analysis identifies new lung cancer susceptibility loci and heterogeneity in genetic susceptibility across histological subtypes". It can be accessed at [this link at Nature genetics](https://www.nature.com/articles/ng.3892) and [this link on PubMed](https://pubmed.ncbi.nlm.nih.gov/28604730/).

Background
------
Genome-Wide Association Studies (GWAS) have previously identified lung cancer susceptiblity loci including CHRNA3, CHRNA5, TERT, HLA (Human Leukocyte Antigen) region, BRCA2 and CHEK2. However, much of the heritability for lung cancer remains unexplained.

PICOTS
------
This is a case-control study performed using existing study genotyping and outcome data.

**Population**: Total 14,803 cases (with diagnosis of lung cancer) and 12,262 controls. These were taken from:
- ATBC study
- CAPUA study
- CARET study
- Canadian screening study
- Copenhagen study
- EAGLE study
- EPIC study
- Liverpool study
- German lung cancer study
- Kentucky lung cancer research initiative
- The Harvard lung cancer study
- Israel study (NICCC-LCA)
- MDACC study
- The Malmo Diet and Cancer Study
- The Multiethnic Cohort
- The Mount-Sinai Hospital-Princess Margaret Study (MSH-PMH)
- The New England Lung Cancer Study (NELCS)
- The Nijmegen Lung cancer study
- Norway study
- NSHDS study
- PLCO study
- Resolucent study
- IARC L2 study
- TLC study
- Washington State University Lung Cancer study
- The Vanderbilt lung cancer study (BioVU)

![Demographics](https://oliver-clark.github.io/images/Screenshot_9.jpg)

**Intervention**: Patients underwent genotyping with OncoArray genotype

**Comparator**: Patients with and without lung cancer

**Outcome**: Association of individual loci with lung cancer

**Setting and Timing**: N/A

Results
------
The study revealed ~21,000 SNPs that are associated with lung cancer with a fixed P-value <1x10^-5. The associations with loci and lung cancer varied according to histological subtypes, as shown in Figure 1. Figure 1 below shows A) overall lung cancer associations, B) associations with Adenocarcinoma, and C) associations with Squamous Cell carcinoma. 
![Figure 1](https://oliver-clark.github.io/images/nihms875731f1.jpg)

These were further narrowed to loci with a fixed P-value <5x10^-8, of which there were 18. 7 were associated with lung cancer overall, 8 were associated with adenocarcinoma, and 3 were associated with squamous cell carcinoma. 

![Table 2](https://oliver-clark.github.io/images/Screenshot_10.jpg)

A further analysis of associated loci was presented in Figure 2. This part was very confusing to me, and it takes a bit of explanation.

![Figure 1](https://oliver-clark.github.io/images/nihms875731f2.jpg)

From my understanding, each graph above represents a different position on the genome, such as 6q27. Each dot on the scatter plots represents a single SNP. The Y-axes represent lung cancer or adenocarcinoma Z-scores. What is the lung cancer Z-score? Within each gene loci are SNPs that have an odds ratio association with lung cancer. The Z-score is the odds ratio of each SNP, standardized to mean 0 and standard deviation of 1 according to the population of SNPs within each gene loci. For example, in Graph A, the SNP rs6920364 is highly associated with lung cancer, and has Z-score ~6, therefore is 6 std above the mean of the odds ratio compared to other SNPs within this same genome position.

The X-axes are even more confusing! They introduce an unfamiliar term - eQTL, which stands for expression of Quantitative Trait Loci. This term evaluates the question - to what extent are variations in these gene loci associated with different expression of a specific mRNA?

I do not fully understand this at the moment but plan to revisit this tomorrow.


Discussion and limitations
------

Take home points
------
